integration_test/arcilator/JIT/verif-simulation.mlir

```diff
@@ -0,0 +1,83 @@
+// RUN: arcilator --run --jit-entry=AdderTest %s | FileCheck %s
+// REQUIRES: arcilator-jit
+
+// CHECK: simulation started
+// CHECK: simulation passed
+
+verif.simulation @AdderTest {} {
+^bb0(%clock: !seq.clock, %init: i1):
+  // Count the first 9001 simulation cycles.
+  %c0_i19 = hw.constant 0 : i19
+  %c1_i19 = hw.constant 1 : i19
+  %c9001_i19 = hw.constant 9001 : i19
+  %count = seq.compreg %0, %clock reset %init, %c0_i19 : i19
+  %done = comb.icmp eq %count, %c9001_i19 : i19
+  %0 = comb.add %count, %c1_i19 : i19
+
+  // Generate inputs to the adder.
+  %1, %2 = func.call @generateAdderInputs(%count) : (i19) -> (i42, i42)
```

Comment:
Strictly speaking this would need to be a module instantiation since the region describes hardware, right? And the function would need to use comb instead of arith.

Reply:
Yeah that's a good point. This happens to survive the Arcilator pipeline unharmed. A DPI call would probably fit better here, to reflect what you would type up in some SV input.